crates/telio-wg/src/wg.rs

```diff
@@ -559,9 +672,49 @@ impl State {
                 let old_state = old.state();
                 let new_state = new.state();
 
+                let old_node_info = self.no_iink_detection_info.get(key);
+                let new_node_info = self.compute_new_node_info(old_node_info, new);
+
+                let old_link_state = old_node_info.map(|i| i.last_link_state);
+                let new_link_state = new_node_info.as_ref().map(|i| i.last_link_state);
+
+                let should_notify_link_state_change: bool =
+                    if matches!(self.no_link_detection, NoLinkDetection::Disabled) {
+                        false
+                    } else if new_link_state.is_none() {
+                        telio_log_warn!("new_link_state is None and it shouldn't");
+                        false
+                    } else {
+                        new_state == PeerState::Connected && old_link_state != new_link_state
+                    };
+
+                // Update info hashmap
+                match new_node_info {
+                    Some(node_info) => {
+                        self.no_iink_detection_info.insert(*key, node_info);
+                    }
+                    None => {
+                        self.no_iink_detection_info.remove(key);
+                    }
+                }
+
                 #[allow(clippy::collapsible_if)]
-                if !old.is_same_event(new) || old_state != new_state {
-                    if !try_send(new_state, new.clone()).await {
+                if !old.is_same_event(new)
+                    || old_state != new_state
+                    || should_notify_link_state_change
+                {
```

Comment:
I think it makes sense to be like this, because is_same_event checks if the endpoint or allowed_ips have changed. And we want to notify this changes too.